boa/src/builtins/value/operations.rs

```diff
@@ -139,3 +207,50 @@ pub fn same_value_non_number(x: &Value, y: &Value) -> bool {
         _ => false,
     }
 }
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness
+#[test]
+fn abstract_equality_comparison() {
+    use crate::{forward, Executor, Realm};
+    let realm = Realm::create();
+    let mut engine = Executor::new(realm);
+
+    assert_eq!(forward(&mut engine, "undefined == undefined"), "true");
+    assert_eq!(forward(&mut engine, "null == null"), "true");
+    assert_eq!(forward(&mut engine, "true == true"), "true");
+    assert_eq!(forward(&mut engine, "false == false"), "true");
+    assert_eq!(forward(&mut engine, "'foo' == 'foo'"), "true");
+    assert_eq!(forward(&mut engine, "0 == 0"), "true");
+    assert_eq!(forward(&mut engine, "+0 == -0"), "true");
+    assert_eq!(forward(&mut engine, "+0 == 0"), "true");
+    assert_eq!(forward(&mut engine, "-0 == 0"), "true");
+    assert_eq!(forward(&mut engine, "0 == false"), "true");
+    assert_eq!(forward(&mut engine, "'' == false"), "true");
+    assert_eq!(forward(&mut engine, "'' == 0"), "true");
+    assert_eq!(forward(&mut engine, "'17' == 17"), "true");
+    assert_eq!(forward(&mut engine, "[1,2] == '1,2'"), "true");
+    assert_eq!(forward(&mut engine, "new String('foo') == 'foo'"), "true");
+    assert_eq!(forward(&mut engine, "null == undefined"), "true");
+    assert_eq!(forward(&mut engine, "undefined == null"), "true");
+    assert_eq!(forward(&mut engine, "null == false"), "false");
```

Comment:
```suggestion
    assert_eq!(forward(&mut engine, "null == false"), "false");
    assert_eq!(forward(&mut engine, "[] == ![]"), "true");
```

Added the thing that originally started the issue.